examples/vmware-cloud-director-machinedeployment.yaml

```diff
@@ -0,0 +1,79 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  # If you change the namespace/name, you must also
+  # adjust the rbac rules
+  name: machine-controller-vcloud-director
+  namespace: kube-system
+type: Opaque
+stringData:
+  password: << VCD_PASSWORD >>
+---
+apiVersion: "cluster.k8s.io/v1alpha1"
+kind: MachineDeployment
+metadata:
+  name: vcloud-director-machinedeployment
+  namespace: kube-system
+spec:
+  paused: false
+  replicas: 1
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxSurge: 1
+      maxUnavailable: 0
+  minReadySeconds: 0
+  selector:
+    matchLabels:
+      foo: bar
+  template:
+    metadata:
+      labels:
+        foo: bar
+    spec:
+      providerSpec:
+        value:
+          sshPublicKeys:
+            - "<< YOUR_PUBLIC_KEY >>"
+          cloudProvider: "vmware-cloud-director"
+          cloudProviderSpec:
+            # Can also be set via the env var 'VCD_USER' on the machine-controller
+            username: "<< VCD_USER >>"
+            # Can also be set via the env var 'VCD_URL' on the machine-controller
+            # example: 'https://your-vcloud-director:8443'. '/api' gets appended automatically
+            url: "<< VCD_URL >>"
+            # Can also be set via the env var 'VCD_PASSWORD' on the machine-controller
+            password:
+              secretKeyRef:
+                namespace: kube-system
+                name: machine-controller-vcloud-director
+                key: password
+            # Can also be set via the env var 'VCD_ORG' on the machine-controller
+            organization: "<< VCD_ORG >>"
+            # Can also be set via the env var 'VCD_VDC' on the machine-controller
+            vdc: "<< VCD_VDC >>"
+            # Can also be set via the env var 'VCD_ALLOW_UNVERIFIED_SSL' on the machine-controller
+            allowInsecure: false
+            # vApp to associate the VM with. This should be created before the machine is created
+            vapp: "<< VCD_VAPP >>"
+            # Name of catalog where the VM template is located
+            catalog: "<< VCD_CATALOG >>"
+            # Name of OS template to be used for the VM
+            template: "<< VCD_TEMPLATE >>"
+            # Direct or routed network that can be used for the VM
+            network: "<< VCD_NETWORK >>"
```

Comment:
I don't see that we define those environment variables in `provider.go`. Is that as expected?

Reply:
My bad -- for some reason I thought those are environment variable names.